app/src/main/java/eu/faircode/netguard/ActivitySettings.java

```diff
@@ -328,8 +336,23 @@ public boolean onPreferenceClick(Preference preference) {
                     final File tmp = new File(getFilesDir(), "hosts.tmp");
                     final File hosts = new File(getFilesDir(), "hosts.txt");
                     EditTextPreference pref_hosts_url = (EditTextPreference) screen.findPreference("hosts_url");
+
+                    URL hostsURL = null;
+                    String hostsPresumedURL = pref_hosts_url.getText();
+
                     try {
-                        new DownloadTask(ActivitySettings.this, new URL(pref_hosts_url.getText()), tmp, new DownloadTask.Listener() {
+                        if (!(hostsPresumedURL.startsWith("http://") || hostsPresumedURL.startsWith("https://"))) {
+                            hostsURL = new URL("http://" + hostsPresumedURL);
+
+                        } else {
+                            hostsURL = new URL(hostsPresumedURL);
+
+                        }
+                    } catch (MalformedURLException ex) {
+                        Toast.makeText(ActivitySettings.this, ex.toString(), Toast.LENGTH_LONG).show();
```

Comment:
What about moving the entire validation path (RFC 3987 + http{s} prefix presence) to HostsTextPreference? I have a working example ready to be pushed.